### ISSUE TITLE
Disable deploy action on PRs

### ISSUE THREAD

**Issue by Peter9192** [eucp-project/atlas#7]
Should only deploy once PR is approved and merged into main.

**Comment by geek-yang** [comment]
About ["Events that trigger workflows"](https://docs.github.com/en/actions/reference/events-that-trigger-workflows) and available syntax ["Workflow syntax for GitHub Actions"](https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#on).

**Comment by geek-yang** [comment]
It sounds like there is no clean option for [Trigger workflow only on pull request MERGE](https://github.community/t/trigger-workflow-only-on-pull-request-merge/17359). The best solution so far seems to be
`on:
  pull_request:
    branches:
      - main
    types: [closed]

jobs:
    cd: // the rest of the code
        if: github.event.pull_request.merged == true
        steps: // the rest of the code`
Any comment about it? @Peter9192

**Comment by Peter9192** [comment]
> It sounds like there is no clean option for [Trigger workflow only on pull request MERGE](https://github.community/t/trigger-workflow-only-on-pull-request-merge/17359).

One of the first answers here suggests: 

```
on:
  push:
    branches:
      - master
```

I think that should work for us (replacing `master` with `main`), right?

**Comment by geek-yang** [comment]
> One of the first answers here suggests:
> 
> ```
> on:
>   push:
>     branches:
>       - master
> ```
> 
> I think that should work for us (replacing `master` with `main`), right?
> 
> Reason: "because a merged pull request always results in a push"

In this case, the cd will only work when we create a pull request and merge it with the main branch. Just wonder if we would like to see if CD works or not before merging the pull request? If not, then this first answer is definitely the best option.

**Comment by Peter9192** [comment]
Oh, I'm not sure if that's possible. Where would we serve the preview site if not on the GH pages branch?

**Comment by Peter9192** [comment]
I suggest we go with the solution that only runs on pushes to main for now. Will make a new issue about testing the deployment
